### PR TITLE
feat(schematics): fail gracefully and warn when project is missing a …

### PR DESCRIPTION
**Pull request by FrozenPandaz**
…format npm script

### Current Behavior

`ng generate` fails hard if there is no format npm script.

### Expected Behavior

image

`ng generate` warn the user that there is no format npm script and that the files were not formatted.

Also refactored wrapIntoFormat into a higher order Rule added `--skip-format` for all schematics that do formatting so they have an option not to see the warning.

### CODE REVIEW

**Comment by FrozenPandaz** [comment]
Fixes #570

**Comment by vsavkin** [comment]
I know the formatting wasn't tested before, but do you think it makes sense to add a unit test to, say, the lib schematic?

File: package.json [hunk 1]
```diff
@@ -75,7 +75,7 @@
   "jest": {
     "modulePathIgnorePatterns": [
       "tmp",
-      "files"
+      "collection/.*/files"
```
Comment (FrozenPandaz) [inline package.json]: Had to make this more specific to not match `format-files.spec.ts`

File: packages/schematics/src/utils/rules/format-files.spec.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  it('should not format files if there is no format npm script', done => {
+    tree.overwrite(
+      'package.json',
+      serializeJson({
+        scripts: {}
+      })
+    );
+    schematicRunner.callRule(formatFiles(), tree).subscribe(result => {
+      expect(schematicRunner.tasks.length).toBe(0);
+      //TODO: test that a warning is emitted.
```
Comment (FrozenPandaz) [inline packages/schematics/src/utils/rules/format-files.spec.ts]: For some reason.. This is very hard to do with `callRule`?

Filed an issue:
https://github.com/angular/devkit/issues/1016

**Pull request merged**